daemon/pkg-mgr.cc

```diff
@@ -159,66 +245,16 @@ _pkg_mgr_event_cb (const char *type, const char *package_name,
 
     g_autofree gchar *app_info = _get_app_info (package_name, res_type, res_version);
 
-    g_autofree gchar *json_file_path
+    g_autofree gchar *model_json
         = g_strdup_printf ("%s/%s/model_description.json", pkg_path, res_type);
-
-    if (!g_file_test (json_file_path, G_FILE_TEST_IS_REGULAR)) {
-      _E ("Failed to find json file '%s'. RPK using ML Service APIs should provide this json file.",
-          json_file_path);
-      return;
-    }
-
-    /* parsing model_description.json */
-    g_autoptr (JsonParser) parser = json_parser_new ();
-    g_autoptr (GError) err = NULL;
-    json_parser_load_from_file (parser, json_file_path, &err);
-    if (err) {
-      _E ("Failed to parse json file '%s': %s", json_file_path, err->message);
-      return;
-    }
-
-    JsonArray *root_array = json_node_get_array (json_parser_get_root (parser));
-    if (!root_array) {
-      _E ("Failed to get root array from json file '%s'", json_file_path);
-      return;
-    }
-
-    guint json_len = json_array_get_length (root_array);
-    if (json_len == 0U) {
-      _E ("Failed to get root array from json file '%s'", json_file_path);
-      return;
-    }
-
-    /* For each model, register it into the database */
-    MLServiceDB &db = MLServiceDB::getInstance ();
-    try {
-      db.connectDB ();
-      for (guint i = 0; i < json_len; ++i) {
-        JsonObjectIter iter;
-        JsonObject *object;
-
-        object = json_array_get_object_element (root_array, i);
-        json_object_iter_init_ordered (&iter, object);
-
-        const gchar *name = json_object_get_string_member (object, "name");
-        const gchar *model = json_object_get_string_member (object, "model");
-        const gchar *description = json_object_get_string_member (object, "description");
-
-        if (!name || !model || !description) {
-          _E ("Failed to get name, model, or description from json file '%s'", json_file_path);
-          return;
-        }
-
-        guint version;
-        db.set_model (name, model, true, description, app_info, &version);
-        _I ("The model with app_info (%s) is registered as version %u", app_info, version);
-      }
-    } catch (const std::exception &e) {
-      _E ("%s", e.what ());
-    }
-
-    db.disconnectDB ();
-
+    g_autofree gchar *pipeline_json
```

Comment:
glib has an API for merging paths. Use https://docs.gtk.org/glib/func.build_filename.html